src/DurableTask.SqlServer/Scripts/logic.sql

```diff
@@ -469,45 +469,70 @@ BEGIN
 
     DECLARE @TaskHub varchar(50) = __SchemaNamePlaceholder__.CurrentTaskHub()
 
-    DECLARE @existingStatus varchar(30) = (
-        SELECT TOP 1 existing.[RuntimeStatus]
-        FROM Instances existing WITH (HOLDLOCK)
-        WHERE [TaskHub] = @TaskHub AND [InstanceID] = @InstanceID
-    )
+    DECLARE @existingStatus varchar(30)
+    DECLARE @existingLockExpiration datetime2(7)
+
+    -- Get the status of an existing orchestration
+    SELECT TOP 1
+        @existingStatus = existing.[RuntimeStatus],
+        @existingLockExpiration = existing.[LockExpiration]
+    FROM Instances existing WITH (HOLDLOCK)
+    WHERE [TaskHub] = @TaskHub AND [InstanceID] = @InstanceID
 
     IF @existingStatus IS NULL
     BEGIN
         ROLLBACK TRANSACTION;
         THROW 50000, 'The instance does not exist.', 1;
     END
-    -- If the instance is already completed, no need to terminate it.
-    IF @existingStatus IN ('Pending', 'Running')
+
+    DECLARE @now datetime2(7) = SYSUTCDATETIME()
+
+    IF @existingStatus IN ('Running', 'Pending')
     BEGIN
-        IF NOT EXISTS (
-            SELECT TOP (1) 1 FROM NewEvents
-            WHERE [TaskHub] = @TaskHub AND [InstanceID] = @InstanceID AND [EventType] = 'ExecutionTerminated'
-        )
+        -- Create a payload to store the reason, if any
+        DECLARE @PayloadID uniqueidentifier = NULL
+        IF @Reason IS NOT NULL
         BEGIN
-            -- Payloads are stored separately from the events
-            DECLARE @PayloadID uniqueidentifier = NULL
-            IF @Reason IS NOT NULL
+            -- Note that we don't use the Reason column for the Reason with terminate events
+            SET @PayloadID = NEWID()
+            INSERT INTO Payloads ([TaskHub], [InstanceID], [PayloadID], [Text])
```

Comment:
The comment in line#468 states that the table order for this sproc should be Instances --> (NewEvents --> Payloads --> NewEvents) but this PR changes the table order to Instances --> (Payloads --> Instances --> NewEvents). 
Does this present a concern for deadlocks, and if not, should we update the comment to reflect the new ordering?

Reply:
Hmm...good observation. Let me take a closer look at this.